crypto/ocsp/ocsp_test.cc

```diff
@@ -1134,3 +1162,85 @@ TEST_P(OCSPHTTPTest, OCSPRequestHTTP) {
     ASSERT_FALSE(ocspResponse);
   }
 }
+
+struct OCSPURLTestVector {
+    const char *ocsp_url;
+    const char *expected_host;
+    const char *expected_port;
+    const char *expected_path;
+    int expected_ssl;
+    int expected_status;
+};
+
+static const OCSPURLTestVector kOCSPURLVectors[] = {
+    // === VALID URLs ===
+    {"http://ocsp.example.com/", "ocsp.example.com", "80", "/", 0,
+     OCSP_URL_PARSE_SUCCESS},
+    // Non-standard port.
+    {"http://ocsp.example.com:8080/", "ocsp.example.com", "8080", "/", 0,
+     OCSP_URL_PARSE_SUCCESS},
+    {"http://ocsp.example.com/ocsp", "ocsp.example.com", "80", "/ocsp", 0,
+     OCSP_URL_PARSE_SUCCESS},
+    // Port for HTTPS is 443.
+    {"https://ocsp.example.com/", "ocsp.example.com", "443", "/", 1,
+     OCSP_URL_PARSE_SUCCESS},
+    // Empty path, the default should be "/".
+    {"https://ocsp.example.com", "ocsp.example.com", "443", "/", 1,
+     OCSP_URL_PARSE_SUCCESS},
+    // Parsing an IPv6 host address.
+    {"http://[2001:db8::1]/ocsp", "2001:db8::1", "80", "/ocsp", 0,
+     OCSP_URL_PARSE_SUCCESS},
+    {"https://[2001:db8::1]/ocsp", "2001:db8::1", "443", "/ocsp", 1,
+     OCSP_URL_PARSE_SUCCESS},
+    // Parsing a URL with an invalid port, but OpenSSL still parses the string
+    // where the port is expected.
+    {"http://ocsp.example.com:invalid/", "ocsp.example.com", "invalid", "/", 0,
+     OCSP_URL_PARSE_SUCCESS},
+    // Empty host component.
+    {"http:///ocsp", "", "80", "/ocsp", 0, OCSP_URL_PARSE_SUCCESS},
+
+    // === INVALID URLs ===
```

Comment:
Do you think more fuzz tests are useful for url parsing?

Reply:
Yeah, I've thought about it, but this function doesn't do any validation of the URL and only parses the string based on a `http(s)://{host}:{port}/{path}` format. It wouldn't hurt to fuzz test this, I'm just not quite sure if there's any additional value. But if we're worried about any edge cases I hadn't thought of, let me know and I can look into doing so.